### PR TITLE
Upgrade knex to avoid CVE-2019-10757 

### PR DESCRIPTION
**Pull request by CameronDiver**
This PR replaces https://github.com/balena-io/balena-supervisor/pull/1124

### CODE REVIEW

File: src/proxyvisor.js [hunk 1]
```diff
@@ -693,14 +693,12 @@ export class Proxyvisor {
...
-			devices: this.db
-				.models('dependentDeviceTarget')
-				.select()
-				.map(this.normaliseDependentDeviceTargetFromDB),
+			apps: Promise.resolve(this.db.models('dependentAppTarget').select()).map(
+				this.normaliseDependentAppFromDB,
+			),
+			devices: Promise.resolve(
+				this.db.models('dependentDeviceTarget').select(),
+			).map(this.normaliseDependentDeviceTargetFromDB),
```
Comment (Page-) [inline src/proxyvisor.js:701]:
```suggestion
			devices: Promise.map(
				this.db.models('dependentDeviceTarget').select(),
				this.normaliseDependentDeviceTargetFromDB
			),
```

File: src/proxyvisor.js [hunk 2]
```diff
@@ -693,14 +693,12 @@ export class Proxyvisor {
...
-				.models('dependentAppTarget')
-				.select()
-				.map(this.normaliseDependentAppFromDB),
-			devices: this.db
-				.models('dependentDeviceTarget')
-				.select()
-				.map(this.normaliseDependentDeviceTargetFromDB),
+			apps: Promise.resolve(this.db.models('dependentAppTarget').select()).map(
+				this.normaliseDependentAppFromDB,
+			),
```
Comment (Page-) [inline src/proxyvisor.js:698]:
```suggestion
			apps: Promise.map(
				this.db.models('dependentAppTarget').select(),
				this.normaliseDependentAppFromDB,
			),
```

File: src/proxyvisor.js [hunk 3]
```diff
@@ -510,16 +510,16 @@ export class Proxyvisor {
...
 
 	getCurrentStates() {
 		return Promise.join(
-			this.db
-				.models('dependentApp')
-				.select()
-				.map(this.normaliseDependentAppFromDB),
+			Promise.resolve(this.db.models('dependentApp').select()).map(
+				this.normaliseDependentAppFromDB,
+			),
```
Comment (Page-) [inline src/proxyvisor.js:522]:
```suggestion
			Promise.map(
				this.db.models('dependentApp').select(),
				this.normaliseDependentAppFromDB,
			),
```

File: src/migrations/20181907164000-endpoint-specific-logschannel.js [hunk 4]
```diff
@@ -1,8 +1,9 @@
+const Bluebird = require('bluebird');
 const fs = require('fs');
 const configJsonPath = process.env.CONFIG_MOUNT_POINT;
 
-exports.up = function(knex, Promise) {
-	return new Promise(resolve => {
+exports.up = function(knex) {
+	return new Bluebird(resolve => {
```
Comment (Page-) [inline src/migrations/20181907164000-endpoint-specific-logschannel.js:6]: You could keep this as 
```suggestion
	return new Promise(resolve => {
```
and just return a native promise - they're definitely supported in node 12

File: src/db.ts [hunk 5]
```diff
@@ -27,42 +26,38 @@ export class DB {
...
-			})
-			.then(() => {
-				return this.knex.migrate.latest({
-					directory: path.join(__dirname, 'migrations'),
-				});
-			});
+	public async init(): Promise<void> {
+		try {
+			await this.knex('knex_migrations_lock').update({ is_locked: 0 });
+		} catch (e) {
```
Comment (Page-) [inline src/db.ts:32]:
```suggestion
		} catch {
```

Comment (CameronDiver) [reply src/db.ts:32]: TIL you can omit the parameters, nice.

File: src/migrations/20181907164000-endpoint-specific-logschannel.js (hunk 4 above)
Comment (CameronDiver) [reply src/migrations/20181907164000-endpoint-specific-logschannel.js:6]: This is for the `.tap` lower down. I could convert these to async/await, but I'd rather leave the existing migrations alone tbh.

File: src/db.ts [hunk 6]
```diff
@@ -14,7 +13,7 @@ export type Transaction = Knex.Transaction;
 
 export class DB {
 	private databasePath: string;
-	private knex: Knex;
+	public knex: Knex;
```
Comment (zvin) [inline src/db.ts:16]: NIT: Why make it public ?

File: src/migrations/20181907164000-endpoint-specific-logschannel.js (hunk 4 above)
Comment (Page-) [reply src/migrations/20181907164000-endpoint-specific-logschannel.js:6]: Ah I couldn't see it, fair enough

File: src/db.ts (hunk 6 above)
Comment (Page-) [reply src/db.ts:16]: I also wonder about this @CameronDiver ?

Comment (CameronDiver) [reply src/db.ts]: Ah, leftover debugging. Good catch.

**Pull request merged**